File: games-frontend/src/types/game.ts
export enum GameSlug {
  Minesweeper = "minesweeper",
  Go = "go",
  TicTacToe = "ticTacToe",
  Ragnarocks = "ragnarocks",
}

export interface GameInfo {
  id: string;
  slug: GameSlug;
  name: string;
  description: string;
  max_players: number | null;
  image_url: string | null;
}

File: games-backend/src/features/games/dtos/game.rs
use serde::Serialize;
use sqlx::FromRow;
use uuid::Uuid;

#[derive(Serialize, FromRow)]
#[serde(rename_all = "camelCase")]
pub struct GameDTO {
    pub id: Uuid,
    pub slug: String,
    pub name: String,
    pub description: Option<String>,
    pub max_players: Option<i32>,
    pub image_url: Option<String>,
}
